src/screen/Screen.java

```diff
@@ -37,6 +40,9 @@ public abstract class Screen {
 	/** What kind of screen goes next. */
 	protected Menu menu;
 
+	protected List<Entity> backBufferEntities;
```

Comment:
Double buffering is already implemented on renderer.
I think just one entity container `entities` is enough.